src/niscope/metadata/functions_addon.py

```diff
@@ -4,4 +4,146 @@
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_date'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niScope_init.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Indicates the **datetime** of the last calibration.'
+                },
+                'name': 'datetime',
+                'type': 'ViInt32'
+            }
+        ],
+        'python_name': 'get_self_cal_last_date_and_time',
+        'returns': 'ViStatus'
+    }
+}
+
+functions_additional_get_ext_cal_last_date_and_time = {
+    'FancyGetLastExtCalDateAndTime': {
+        'codegen_method': 'python-only',
+        'documentation': {
+            'description': 'Returns the date and time of the last external calibration performed.',
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_date'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niScope_init.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Indicates the **datetime** of the last calibration.'
+                },
+                'name': 'datetime',
+                'type': 'ViInt32'
+            }
+        ],
+        'python_name': 'get_ext_cal_last_date_and_time',
+        'returns': 'ViStatus'
+    }
+}
 
+functions_additional_get_self_cal_last_temp = {
+    'FancyGetLastSelfCalTemp': {
+        'codegen_method': 'python-only',
+        'python_name': 'get_self_cal_last_temp',
+        'documentation': {
+            'description': 'Returns the **Temperature** in degrees Celsius during the last self calibration procedure.',
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_temp'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'The instrument handle you obtain from niScope_init that identifies a particular instrument session.'
```

Comment:
Keep the docstring for this parameter consistent among all the newly added methods.